tests/container/vertx/src/test/java/org/hawkular/apm/tests/client/vertx/ClientVertxEventBusTest.java

```diff
@@ -72,14 +73,6 @@ public void handle(AsyncResult<Message<String>> message) {
             }
         });
 
-        try {
-            synchronized (this) {
-                wait(2000);
-            }
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-
         mc.unregister();
```

Comment:
Forgot to reply to this one (valid also for the next): one solution could be to introduce a CountDownLatch in a second consumer. This second consumer does the countDown and releases the latch. The latch would wait before the "unregister" call. 

Assuming we can register a consumer on the test itself.